scripts/portable-sbob.py

```
@@ -0,0 +1,151 @@
+            for o in s["opens"]:
+                cur = by.setdefault(o["path"], o)
+                cur["flags"] = sorted(set(cur.get("flags") or []) | set(o.get("flags") or []))
+            s["opens"] = sorted(by.values(), key=lambda o: o["path"])
+
+        print(f"  {Path(f).name}")
+        for n in notes:
+            print(f"      {n}")
+        if not notes:
+            print("      (already portable)")
+        if not args.dry_run:
+            yaml.dump(d, open(f, "w"), sort_keys=False, width=4096, allow_unicode=True,
```

Ruff (0.16.0)

[warning] Use a context manager for opening files

(SIM115)